src/api/exchange/ExchangeApi.ts

```diff
@@ -109,6 +121,37 @@ export class ExchangeApiImpl extends DepositApiImpl implements ExchangeApi {
     return decodeAuctionElements(encodedOrders)
   }
 
+  public async getOrdersPaginated({
+    userAddress,
+    networkId,
+    offset,
+    pageSize = DEFAULT_ORDERS_PAGE_SIZE,
+  }: GetOrdersPaginatedParams): Promise<GetOrdersPaginatedResult> {
+    const contract = await this._getContract(networkId)
+
+    log(
+      `[ExchangeApiImpl] Getting Orders Paginated for account ${userAddress} with offset ${offset} and pageSize ${pageSize}`,
+    )
+
+    // query 1 more than required to check whether there's a next page
+    const encodedOrders = await contract.methods.getEncodedUserOrdersPaginated(userAddress, offset, pageSize + 1).call()
+
+    // is null if Contract returns empty bytes
+    if (!encodedOrders) return { orders: [] }
+
+    const orders = decodeAuctionElements(encodedOrders, offset)
+
+    if (orders.length <= pageSize) {
+      // no more pages left, indicate by not returning `nextIndex`
+      return { orders }
+    } else {
+      // there is at least 1 item in the next page,
+      // pop the extra element, get its id as nextIndex
+      const nextIndex = Number((orders.pop() as AuctionElement).id)
```

Comment:
nit: pls consider do just one thing per line for clarity. In that line, you are doing a pop (mutate the array to discard the last element) and then getting it's id.

Reply:
Sure